File: programmers/Lv_2/JeongJunKwon/카펫.java
class Solution {
    public int[] solution(int brown, int red) {
        int[] answer = new int[2];
        int width = 0;
        int height = 0;
        int total = 0;
        while(total != brown) {
            height++;
            if(red % height != 0)
                continue;
            width = (red / height);
            total = width * 2 + height * 2 + 4;
        }
        
        answer[0] = width + 2;
        answer[1] = height + 2;
        return answer;
    }
}

File: programmers/Lv_2/HwangJunghoon/카펫.py
def solution(brown, red):
    answer = []
    ans=[]
    num=brown+red
    for i in range(num+1, 0, -1):
        if num%i==0:
            ans.append(i)
            if num//i==i:
                ans.append(i)
    for j in range(1, len(ans)):
        x,y=ans[j-1],ans[len(ans)-j]
        if (ans[j-1]-2)*(ans[len(ans)-j]-2)==red:
            answer.append(x)
            answer.append(y)
            break
    return answer